lib/parallel_tests/grouper.rb

```diff
@@ -28,7 +28,39 @@ def in_even_groups_by_size(items, num_groups, options = {})
           raise 'Number of isolated processes must be less than total the number of processes'
         end
 
-        if isolate_count >= 1
+        if options[:specify_groups]
+          specify_test_process_groups = options[:specify_groups].split('|')
+          all_specified_tests = specify_test_process_groups.map { |group| group.split(',') }.flatten
+          if specify_test_process_groups.count > num_groups
+            raise 'Number of processes separated by pipe must be less than or equal to the total number of processes'
+          end
+
+          specified_items_found, items = items.partition do |item, _size|
+            all_specified_tests.any? { |specified_spec| item == specified_spec }
+          end
+
+          specified_specs_not_found = all_specified_tests - specified_items_found.map(&:first)
+          if specified_specs_not_found.any?
+            raise "Could not find #{specified_specs_not_found} from --specify-groups in the main selected files & folders"
+          end
+
+          if specify_test_process_groups.count == num_groups && items.flatten.any?
+            raise "The number of groups in --specify-groups matches the number of groups from -n but there were other specs found in the main selected files & folders not specified in --specify-groups. Make sure -n is larger than the number of processes in --specify-groups if there are other specs that need to be run. The specs that aren't run: #{items.map(&:first)}"
+          end
+
+          specify_test_process_groups.each_with_index do |specify_test_process, i|
+            groups[i] = specify_test_process.split(',')
+          end
+
+          return groups if specify_test_process_groups.count == num_groups
+
+          regular_group_starting_process = specify_test_process_groups.count
+          group_features_by_size(items_to_group(items), groups[regular_group_starting_process..-1])
+          # Don't sort all the groups, only sort the ones not specified in specify_groups
+          sorted_groups = groups[regular_group_starting_process..-1].map { |g| g[:items].sort }
+          groups[regular_group_starting_process..-1] = sorted_groups
```

Comment:
```suggestion
          groups[regular_group_starting_process..-1].each { |g| g[:items].sort! }
```

Reply:
I don't think this works because the map was getting rid of the item/size hash and was making it just an array of arrays
<img width="1087" alt="02-08-2021 14 43 21_Oq2tnf" src="https://user-images.githubusercontent.com/12533693/107278956-0b87a680-6a1c-11eb-85d6-0868d5dd3135.png">
<img width="1204" alt="02-08-2021 14 41 28_miWC8x" src="https://user-images.githubusercontent.com/12533693/107278963-0cb8d380-6a1c-11eb-8776-dae0955a9165.png">

Reply:
hmmm not a big fan that we have mixed types in the groups then, so maybe better to have the same type in there from the beginning ?

Reply:
maybe start by extracting the range to a variable and I'll take a closer look when you think it's ready again

Reply:
I was just copying how the other groups do this step, at the end of their blocks

instead of 

```ruby
sorted_groups = groups[regular_group_starting_process..-1].map { |g| g[:items].sort }
groups[regular_group_starting_process..-1] = sorted_groups
```


it could be 1 line
```ruby
groups[regular_group_starting_process..-1].map! { |g| g[:items].sort }
```